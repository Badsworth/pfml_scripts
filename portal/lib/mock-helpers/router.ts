/**
 * @file Common next/router mocks for use in Storybook and Tests
 */
import Router from "next/router";
import createMockFn from "./createMockFn";

export interface MockRouterEvent {
  name: string;
  callback: RouterEventCallback;
}

export const mockRouterEvents: MockRouterEvent[] = [];

type RouterEventCallback = (...args: unknown[]) => unknown;

// @ts-expect-error Router mock is missing properties we don't currently reference
export const mockRouter = {
  locale: "en-US",
  route: "/",
  pathname: "",
  query: {},
  asPath: "",
  back: createMockFn(),
  beforePopState: createMockFn(),
  prefetch: createMockFn(() => Promise.resolve()),
  push: createMockFn(),
  replace: createMockFn(),
  reload: createMockFn(),
  events: {
    on: (name: string, callback: RouterEventCallback) => {
      mockRouterEvents.push({ name, callback });
    },
    off: createMockFn(),
  },
  isFallback: false,
} as typeof Router.router;

export function useRouter() {
  return mockRouter;
}
