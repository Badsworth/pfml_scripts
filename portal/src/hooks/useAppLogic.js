/* eslint sort-keys: ["error", "asc"] */
import useAppErrorsLogic from "./useAppErrorsLogic";
import useAuthLogic from "./useAuthLogic";
import useBenefitsApplicationsLogic from "./useBenefitsApplicationsLogic";
import useClaimsLogic from "./useClaimsLogic";
import useDocumentsLogic from "./useDocumentsLogic";
import useEmployersLogic from "./useEmployersLogic";
import useFlagsLogic from "./useFeatureFlagsLogic";
import useOtherLeavesLogic from "./useOtherLeavesLogic";
import usePortalFlow from "./usePortalFlow";
import useUsersLogic from "./useUsersLogic";

const useAppLogic = () => {
  // Utility to determine application flow
  const portalFlow = usePortalFlow();

  // State representing currently visible errors and warnings
  const { appErrors, ...appErrorsLogic } = useAppErrorsLogic({ portalFlow });
  const auth = useAuthLogic({ appErrorsLogic, portalFlow });

  // State representing the Portal's user object.
  // Initialize to empty user but will be populated upon the first API call
  // to fetch the user (or create the user on their first login)
  const users = useUsersLogic({
    appErrorsLogic,
    isLoggedIn: auth.isLoggedIn,
    portalFlow,
  });

  // user will be eventually set through a `login` method
  // const [user, setUser] = useState();
  // TODO (CP-885): remove user from configs and write login method
  const benefitsApplications = useBenefitsApplicationsLogic({
    appErrorsLogic,
    portalFlow,
    user: users.user,
  });

  const claims = useClaimsLogic({
    appErrorsLogic,
  });

  const documents = useDocumentsLogic({
    appErrorsLogic,
    portalFlow,
  });

  const employers = useEmployersLogic({
    appErrorsLogic,
    portalFlow,
    setUser: users.setUser,
  });

  const otherLeaves = useOtherLeavesLogic({ appErrorsLogic });
  const featureFlags = useFlagsLogic({ appErrorsLogic });

  return {
    appErrors,
    auth,
    benefitsApplications,
    catchError: appErrorsLogic.catchError,
    claims,
    // TODO (CP-886): remove once all API calls are behind appLogic
    clearErrors: appErrorsLogic.clearErrors,
    documents,
    employers,
    otherLeaves,
    portalFlow,
    setAppErrors: appErrorsLogic.setAppErrors,
    users,
    featureFlags,
  };
};

export default useAppLogic;
