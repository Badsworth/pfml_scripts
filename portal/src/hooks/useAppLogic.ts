/* eslint sort-keys: ["error", "asc"] */
import useApplicationImportsLogic from "./useApplicationImportsLogic";
import useAuthLogic from "./useAuthLogic";
import useBenefitYearsLogic from "./useBenefitYearsLogic";
import useBenefitsApplicationsLogic from "./useBenefitsApplicationsLogic";
import useClaimsLogic from "./useClaimsLogic";
import useDocumentsLogic from "./useDocumentsLogic";
import useEmployersLogic from "./useEmployersLogic";
import useErrorsLogic from "./useErrorsLogic";
import useFeatureFlagsLogic from "./useFeatureFlagsLogic";
import useHolidaysLogic from "./useHolidaysLogic";
import usePaymentsLogic from "./usePaymentsLogic";
import usePortalFlow from "./usePortalFlow";
import useUsersLogic from "./useUsersLogic";

const useAppLogic = () => {
  // Utility to determine application flow
  const portalFlow = usePortalFlow();

  // State representing currently visible errors and warnings
  const errorsLogic = useErrorsLogic({ portalFlow });

  const applicationImports = useApplicationImportsLogic({
    errorsLogic,
    portalFlow,
  });
  const auth = useAuthLogic({ errorsLogic, portalFlow });

  // State representing the Portal's user object.
  // Initialize to empty user but will be populated upon the first API call
  // to fetch the user (or create the user on their first login)
  const users = useUsersLogic({
    errorsLogic,
    isLoggedIn: !!auth.isLoggedIn,
    portalFlow,
  });

  const benefitYears = useBenefitYearsLogic({ errorsLogic });

  const benefitsApplications = useBenefitsApplicationsLogic({
    errorsLogic,
    portalFlow,
  });

  const claims = useClaimsLogic({
    errorsLogic,
    portalFlow,
  });

  const documents = useDocumentsLogic({
    errorsLogic,
  });

  const employers = useEmployersLogic({
    clearClaims: claims.clearClaims,
    errorsLogic,
    portalFlow,
    setUser: users.setUser,
  });

  const featureFlags = useFeatureFlagsLogic();

  const holidays = useHolidaysLogic({
    errorsLogic,
  });

  const payments = usePaymentsLogic({
    errorsLogic,
  });

  return {
    // `_errorsLogic` should not be used except for testing
    _errorsLogic: errorsLogic,
    applicationImports,
    auth,
    benefitYears,
    benefitsApplications,
    catchError: errorsLogic.catchError,
    claims,
    // TODO (CP-886): remove once all API calls are behind appLogic
    clearErrors: errorsLogic.clearErrors,
    clearRequiredFieldErrors: errorsLogic.clearRequiredFieldErrors,
    documents,
    employers,
    errors: errorsLogic.errors,
    featureFlags,
    holidays,
    payments,
    portalFlow,
    setErrors: errorsLogic.setErrors,
    users,
  };
};

export default useAppLogic;
export type AppLogic = ReturnType<typeof useAppLogic>;
