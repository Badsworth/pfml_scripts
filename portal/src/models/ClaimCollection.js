import BaseCollection from "./BaseCollection";
import { ClaimStatus } from "./Claim";

/** @typedef {import('./Claim').default} Claim */

export default class ClaimCollection extends BaseCollection {
  get idProperty() {
    return "application_id";
  }

  /**
   * @returns {Claim[]} Returns all claims with an "In progress" status
   */
  get inProgress() {
    return this.items.filter((item) => item.status === ClaimStatus.started);
  }

  /**
   * @returns {Claim[]} Returns all claims submitted to the API, including
   * those that made it to the Claims Processing System
   */
  get submitted() {
    return this.items.filter((item) => item.isCompleted || item.isSubmitted);
  }
}
