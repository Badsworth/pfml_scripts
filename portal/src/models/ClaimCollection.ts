import BaseCollection from "./BaseCollection";
import Claim from "./Claim";

export default class ClaimCollection extends BaseCollection<Claim> {
  get idProperty() {
    return "fineos_absence_id";
  }
}
