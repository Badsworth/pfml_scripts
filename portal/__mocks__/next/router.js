import { mockRouter, mockRouterEvents } from "../../lib/mock-helpers/router";
export * from "../../lib/mock-helpers/router";

afterEach(() => {
  mockRouter.pathname = "";

  // Clear mock router events
  mockRouterEvents.length = 0;
});
