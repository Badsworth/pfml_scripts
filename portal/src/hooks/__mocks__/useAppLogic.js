import Claim from "../../models/Claim";
import ClaimCollection from "../../models/ClaimCollection";
import { uniqueId } from "lodash";

export default jest.fn(() => ({
  auth: {
    createAccount: jest.fn(),
    forgotPassword: jest.fn(),
    login: jest.fn(),
  },
  claims: new ClaimCollection(),
  createClaim: jest.fn(() => new Claim({ application_id: uniqueId() })),
  loadClaims: jest.fn(),
  submitClaim: jest.fn(),
  updateClaim: jest.fn(),
}));
