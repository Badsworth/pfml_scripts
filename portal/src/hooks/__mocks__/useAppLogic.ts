import {
  BenefitsApplicationDocument,
  ClaimDocument,
} from "../../models/Document";
import User, { UserLeaveAdministrator } from "../../models/User";
import ApiResourceCollection from "../../models/ApiResourceCollection";
import BenefitsApplication from "../../models/BenefitsApplication";
import Claim from "../../models/Claim";
import ClaimDetail from "../../models/ClaimDetail";
import EmployerClaim from "../../models/EmployerClaim";
import Flag from "../../models/Flag";
import { Payment } from "src/models/Payment";
import { uniqueId } from "lodash";

export default jest.fn(() => ({
  errors: [],
  auth: {
    createAccount: jest.fn(),
    createEmployerAccount: jest.fn(),
    forgotPassword: jest.fn(),
    isLoggedIn: true,
    login: jest.fn(),
    logout: jest.fn(),
    requireLogin: jest.fn(),
    resendForgotPasswordCode: jest.fn(),
    resendVerifyAccountCode: jest.fn(),
    resetPassword: jest.fn(),
    verifyAccount: jest.fn(),
  },
  catchError: jest.fn(),
  benefitsApplications: {
    benefitsApplications: new ApiResourceCollection<BenefitsApplication>(
      "application_id"
    ),
    complete: jest.fn(),
    create: jest.fn(
      () => new BenefitsApplication({ application_id: uniqueId() })
    ),
    get: jest.fn(),
    hasLoadedBenefitsApplicationAndWarnings: jest.fn().mockReturnValue(true),
    hasLoadedAll: false,
    load: jest.fn(),
    loadAll: jest.fn(),
    submit: jest.fn(),
    submitPaymentPreference: jest.fn(),
    update: jest.fn(),
    warningsLists: {},
  },
  claims: {
    claimDetail: new ClaimDetail(),
    claims: new ApiResourceCollection<Claim>("fineos_absence_id"),
    clearClaims: jest.fn(),
    isLoadingClaims: null,
    isLoadingClaimDetail: null,
    loadClaimDetail: jest.fn(),
    paginationMeta: {},
    loadPage: jest.fn(),
  },
  clearErrors: jest.fn(),
  clearRequiredFieldErrors: jest.fn(),
  documents: {
    attach: jest.fn((application_id, files) => {
      const uploadPromises: Array<Promise<{ success: boolean }>> = [];
      for (let i = 0; i < files.length; i++) {
        uploadPromises.push(Promise.resolve({ success: true }));
      }
      return uploadPromises;
    }),
    hasLoadedClaimDocuments: jest.fn(),
    documents: new ApiResourceCollection<BenefitsApplicationDocument>(
      "fineos_document_id"
    ),
    download: jest.fn(),
    loadAll: jest.fn(),
  },
  employers: {
    addEmployer: jest.fn(),
    downloadDocument: jest.fn(() => new Blob()),
    loadClaim: jest.fn(
      () =>
        new EmployerClaim({
          absence_periods: [],
          fineos_absence_id: "NTN-111-ABS-01",
        })
    ),
    loadDocuments: jest.fn(
      () => new ApiResourceCollection<ClaimDocument>("fineos_document_id")
    ),
    loadWithholding: jest.fn(() => ({ filing_period: "2011-11-20" })),
    submitClaimReview: jest.fn(),
    submitWithholding: jest.fn(),
  },
  payments: {
    loadPayments: jest.fn(),
    loadedPaymentsData: new Payment(),
    hasLoadedPayments: true,
    isLoadingPayments: false,
  },
  portalFlow: {
    getNextPageRoute: jest.fn(),
    goTo: jest.fn(),
    goToNextPage: jest.fn(),
    goToPageFor: jest.fn(),
    pathname: "/mock-portalFlow-pathname",
  },
  otherLeaves: {
    removeEmployerBenefit: jest.fn(() => true),
    removeOtherIncome: jest.fn(() => true),
    removePreviousLeave: jest.fn(() => true),
  },
  setErrors: jest.fn(),
  updateUser: jest.fn(),
  user: new User({
    user_id: "mock_user_id",
    consented_to_data_sharing: true,
    email_address: "ali@miau.com",
    auth_id: "cognito_123",
    roles: [],
    user_leave_administrators: [],
  }),
  users: {
    loadUser: jest.fn(),
    requireUserConsentToDataAgreement: jest.fn(),
    requireUserRole: jest.fn(),
    updateUser: jest.fn(),
    user: new User({
      auth_id: "mock_auth_id",
      user_id: "mock_user_id",
      consented_to_data_sharing: true,
      user_leave_administrators: [
        // not verified, but can be
        new UserLeaveAdministrator({
          employer_dba: "Book Bindings 'R Us",
          employer_fein: "**-***1823",
          employer_id: "dda903f-f093f-ff900",
          has_fineos_registration: true,
          has_verification_data: true,
          verified: false,
        }),
        // already verified
        new UserLeaveAdministrator({
          employer_dba: "Knitting Castle",
          employer_fein: "**-***3443",
          employer_id: "dda930f-93jfk-iej08",
          has_fineos_registration: true,
          has_verification_data: true,
          verified: true,
        }),
        // not verified and cannot be verified
        new UserLeaveAdministrator({
          employer_dba: "Tomato Touchdown",
          employer_fein: "**-***7192",
          employer_id: "io19fj9-00jjf-uiw3r",
          has_fineos_registration: true,
          has_verification_data: false,
          verified: false,
        }),
      ],
    }),
  },
  featureFlags: {
    flags: [
      new Flag({
        enabled: false,
        name: "maintenance",
        options: {
          page_routes: ["/*"],
        },
        start: null,
        end: null,
      }),
    ],
    getFlag: jest.fn(() => {
      return new Flag({
        enabled: false,
        name: "maintenance",
        options: {
          page_routes: ["/*"],
        },
        start: null,
        end: null,
      });
    }),
    loadFlags: jest.fn(),
  },
}));
