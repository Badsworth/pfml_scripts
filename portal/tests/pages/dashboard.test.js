import { MockClaimBuilder, renderWithAppLogic, testHook } from "../test-utils";
import ClaimCollection from "../../src/models/ClaimCollection";
import Dashboard from "../../src/pages/dashboard";
import User from "../../src/models/User";
import { mockRouter } from "next/router";
import routes from "../../src/routes";
import useAppLogic from "../../src/hooks/useAppLogic";

function render(customProps = {}, options = { claims: [] }) {
  let appLogic;
  mockRouter.pathname = routes.applications.dashboard;

  testHook(() => {
    appLogic = useAppLogic();
    appLogic.users.user = new User({ consented_to_data_sharing: true });
    appLogic.claims.claims = new ClaimCollection(options.claims);
    appLogic.claims.hasLoadedAll = true;
  });

  const goToPageForSpy = jest.spyOn(appLogic.portalFlow, "goToPageFor");
  const { wrapper } = renderWithAppLogic(Dashboard, {
    props: { appLogic, ...customProps },
  });

  return { appLogic, goToPageForSpy, wrapper };
}

describe("Dashboard", () => {
  it("renders dashboard content", () => {
    const { wrapper } = render();

    expect(wrapper).toMatchSnapshot();
    wrapper.find("Trans").forEach((trans) => {
      expect(trans.dive()).toMatchSnapshot();
    });
  });

  it("redirects to Dashboard when user has applications and just logged in", () => {
    const claims = [new MockClaimBuilder().create()];

    const { goToPageForSpy } = render(
      {
        query: {
          "logged-in": "true",
        },
      },
      { claims }
    );

    expect(goToPageForSpy).toHaveBeenCalledWith("SHOW_APPLICATIONS");
  });

  it("does not redirect to Dashboard when user has applications but logged-in query isn't set", () => {
    const claims = [new MockClaimBuilder().create()];

    const { goToPageForSpy } = render({}, { claims });

    expect(goToPageForSpy).not.toHaveBeenCalled();
  });
});
