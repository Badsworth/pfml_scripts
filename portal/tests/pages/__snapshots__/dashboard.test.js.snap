// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard renders dashboard content 1`] = `
<Fragment>
  <DashboardNavigation
    activeHref="/dashboard"
  />
  <Title>
    Get ready to apply
  </Title>
  <Alert
    className="margin-bottom-3"
    heading="Only some people can apply online for now"
    neutral={true}
    state="info"
  >
    <Heading
      className="margin-top-3"
      level="3"
    >
      <SvgIconLaptop
        className="margin-right-1 text-secondary text-middle"
        fill="currentColor"
        height={20}
        width={20}
      />
      Apply online
    </Heading>
    <Trans
      components={
        Object {
          "li": <li />,
          "ul": <ul
            className="usa-list"
          />,
        }
      }
      i18nKey="pages.dashboard.alertOnline"
    />
    <Heading
      level="3"
    >
      <SvgIconPhone
        className="margin-right-1 text-secondary text-middle"
        fill="currentColor"
        height={20}
        width={20}
      />
      Apply by phone
    </Heading>
    <Trans
      components={
        Object {
          "benefits-timeline-link": <a
            href="https://www.mass.gov/info-details/paid-family-and-medical-leave-pfml-benefits-timeline#december-2,-2020-"
          />,
          "contact-center-phone-link": <a
            href="tel:(833) 344‑7365"
          />,
          "li": <li />,
          "ul": <ul
            className="usa-list"
          />,
        }
      }
      i18nKey="pages.dashboard.alertPhone"
    />
  </Alert>
  <div
    className="measure-6"
  >
    <Heading
      level="2"
    >
      <FontAwesomeIcon
        border={false}
        className="margin-right-1 text-secondary"
        fixedWidth={true}
        flip={null}
        icon={
          Object {
            "icon": Array [
              576,
              512,
              Array [],
              "f086",
              "M532 386.2c27.5-27.1 44-61.1 44-98.2 0-80-76.5-146.1-176.2-157.9C368.3 72.5 294.3 32 208 32 93.1 32 0 103.6 0 192c0 37 16.5 71 44 98.2-15.3 30.7-37.3 54.5-37.7 54.9-6.3 6.7-8.1 16.5-4.4 25 3.6 8.5 12 14 21.2 14 53.5 0 96.7-20.2 125.2-38.8 9.2 2.1 18.7 3.7 28.4 4.9C208.1 407.6 281.8 448 368 448c20.8 0 40.8-2.4 59.8-6.8C456.3 459.7 499.4 480 553 480c9.2 0 17.5-5.5 21.2-14 3.6-8.5 1.9-18.3-4.4-25-.4-.3-22.5-24.1-37.8-54.8zm-392.8-92.3L122.1 305c-14.1 9.1-28.5 16.3-43.1 21.4 2.7-4.7 5.4-9.7 8-14.8l15.5-31.1L77.7 256C64.2 242.6 48 220.7 48 192c0-60.7 73.3-112 160-112s160 51.3 160 112-73.3 112-160 112c-16.5 0-33-1.9-49-5.6l-19.8-4.5zM498.3 352l-24.7 24.4 15.5 31.1c2.6 5.1 5.3 10.1 8 14.8-14.6-5.1-29-12.3-43.1-21.4l-17.1-11.1-19.9 4.6c-16 3.7-32.5 5.6-49 5.6-54 0-102.2-20.1-131.3-49.7C338 339.5 416 272.9 416 192c0-3.4-.4-6.7-.7-10C479.7 196.5 528 238.8 528 288c0 28.7-16.2 50.6-29.7 64z",
            ],
            "iconName": "comments",
            "prefix": "far",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size="lg"
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
      1. Tell your employer that you need to take paid leave
    </Heading>
    <Trans
      i18nKey="pages.dashboard.stepOne"
    />
    <Heading
      level="2"
    >
      <FontAwesomeIcon
        border={false}
        className="margin-right-1 text-secondary"
        fixedWidth={true}
        flip={null}
        icon={
          Object {
            "icon": Array [
              384,
              512,
              Array [],
              "f15b",
              "M369.9 97.9L286 14C277 5 264.8-.1 252.1-.1H48C21.5 0 0 21.5 0 48v416c0 26.5 21.5 48 48 48h288c26.5 0 48-21.5 48-48V131.9c0-12.7-5.1-25-14.1-34zM332.1 128H256V51.9l76.1 76.1zM48 464V48h160v104c0 13.3 10.7 24 24 24h104v288H48z",
            ],
            "iconName": "file",
            "prefix": "far",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size="lg"
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
      2. Get documentation that supports your leave request
    </Heading>
    <Trans
      components={
        Object {
          "healthcare-provider-form-link": <a
            href="https://www.mass.gov/hcp-form"
          />,
        }
      }
      i18nKey="pages.dashboard.stepTwo"
    />
    <Heading
      level="2"
    >
      <FontAwesomeIcon
        border={false}
        className="margin-right-1 text-secondary"
        fixedWidth={true}
        flip={null}
        icon={
          Object {
            "icon": Array [
              576,
              512,
              Array [],
              "f044",
              "M402.3 344.9l32-32c5-5 13.7-1.5 13.7 5.7V464c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V112c0-26.5 21.5-48 48-48h273.5c7.1 0 10.7 8.6 5.7 13.7l-32 32c-1.5 1.5-3.5 2.3-5.7 2.3H48v352h352V350.5c0-2.1.8-4.1 2.3-5.6zm156.6-201.8L296.3 405.7l-90.4 10c-26.2 2.9-48.5-19.2-45.6-45.6l10-90.4L432.9 17.1c22.9-22.9 59.9-22.9 82.7 0l43.2 43.2c22.9 22.9 22.9 60 .1 82.8zM460.1 174L402 115.9 216.2 301.8l-7.3 65.3 65.3-7.3L460.1 174zm64.8-79.7l-43.2-43.2c-4.1-4.1-10.8-4.1-14.8 0L436 82l58.1 58.1 30.9-30.9c4-4.2 4-10.8-.1-14.9z",
            ],
            "iconName": "edit",
            "prefix": "far",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size="lg"
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
      3. Apply
    </Heading>
    <Trans
      components={
        Object {
          "contact-center-phone-link": <a
            href="tel:(833) 344‑7365"
          />,
          "li": <li />,
          "medical-bonding-link": <a
            href="https://www.mass.gov/pfml-medical-bonding"
          />,
          "ul": <ul
            className="usa-list"
          />,
        }
      }
      i18nKey="pages.dashboard.stepThree"
    />
    <ButtonLink
      className="margin-top-3 margin-bottom-8"
      href="/applications/start"
    >
      Create an application
    </ButtonLink>
  </div>
</Fragment>
`;

exports[`Dashboard renders dashboard content 2`] = `
Array [
  <p
    key="p-0"
  >
    If you are currently employed in Massachusetts but not self-employed, you can apply online or by phone for the following types of paid leave:
  </p>,
  <ul
    className="usa-list"
    key="1"
  >
    <li
      key="0"
    >
      Medical leave due to your own illness, injury, or pregnancy
    </li>
    <li
      key="1"
    >
      Family leave to bond with your child after birth, adoption, or foster placement — whether you are applying before or after the child arrives
    </li>
  </ul>,
]
`;

exports[`Dashboard renders dashboard content 3`] = `
Array [
  <p
    key="p-0"
  >
    Apply by calling the Department of Family and Medical Leave Contact Center at 
    <a
      href="tel:(833) 344‑7365"
      key="1"
    >
      (833) 344‑7365
    </a>
     if 
    <strong
      key="strong-3"
    >
      any
    </strong>
     of the following are true:
  </p>,
  <ul
    className="usa-list"
    key="1"
  >
    <li
      key="0"
    >
      You are self-employed or unemployed and you are applying for paid leave
    </li>
    <li
      key="1"
    >
      You need paid family leave to care for a family member who serves in the armed forces
    </li>
    <li
      key="2"
    >
      You need paid family leave to manage family affairs when a family member is on active duty in the armed forces
    </li>
  </ul>,
]
`;

exports[`Dashboard renders dashboard content 4`] = `
Array [
  <p
    key="p-0"
  >
    If you can, tell your employer at least 30 days before your leave begins. If you need to take leave right away, tell your employer as soon as possible.
  </p>,
  <p
    key="p-1"
  >
    Once you tell your employer, you have the right to apply and your job is protected. Make a note of when you notified your employer. You will need to provide this date in your leave application.
  </p>,
]
`;

exports[`Dashboard renders dashboard content 5`] = `
Array [
  <p
    key="p-0"
  >
    For medical leave your health care provider must complete the 
    <a
      href="https://www.mass.gov/hcp-form"
      key="1"
    >
      PFML Certification of a Serious Health Condition
    </a>
    .
  </p>,
  <p
    key="p-1"
  >
    For family leave to bond with your child after birth, foster placement, or adoption you need a document that confirms your child’s date of birth or placement.
  </p>,
  <p
    key="p-2"
  >
    You can apply before your child is born or arrives in your home. You will need to provide proof of birth or placement for your application to be approved.
  </p>,
]
`;

exports[`Dashboard renders dashboard content 6`] = `
Array [
  <p
    key="p-0"
  >
    Applying takes around 15 minutes. Your information will save as you go, so you can finish your application later if you need to.
  </p>,
  <p
    key="p-1"
  >
    If you give birth and plan to take both pregnancy-related medical leave and family leave to bond with your newborn, you should apply for medical leave first. Family leave to bond with your child can be 
    <a
      href="https://www.mass.gov/pfml-medical-bonding"
      key="1"
    >
      easily added to your claim
    </a>
     by calling our Contact Center at 
    <a
      href="tel:(833) 344‑7365"
      key="3"
    >
      (833) 344‑7365
    </a>
    .
  </p>,
  <p
    key="p-2"
  >
    You need to create multiple leave applications if you are:
  </p>,
  <ul
    className="usa-list"
    key="3"
  >
    <li
      key="0"
    >
      Taking leave from multiple employers.
    </li>
    <li
      key="1"
    >
      Taking time off in uneven blocks of time (intermittent leave), 
      <strong
        key="strong-1"
      >
        and
      </strong>
       taking time off completely or on a reduced schedule. You’ll need a separate application for the intermittent leave.
    </li>
  </ul>,
]
`;
